src/android/toga_android/app.py

```diff
@@ -7,9 +7,21 @@ class MainWindow(Window):
     pass
 
 
-class TogaApp:
+# The `IPythonApp` interface in Java allows Python code to
+# run on Android activity lifecycle hooks such as `onCreate()`.
+IPythonApp = JavaInterface('org/beeware/android/IPythonApp')
+
+
+class TogaApp(IPythonApp):
     def __init__(self, app):
+        super().__init__()
         self._interface = app
+        activity_class = JavaClass('org/beeware/android/MainActivity')
```

Comment:
Convention from iOS/macOS would be to declare this as `MainActivity`, and do so either at the module level, or in a `libs` module that contains all the binding declarations.